lib/tag-finder.coffee

```diff
@@ -10,23 +11,38 @@ class TagFinder
   constructor: (@editor) ->
     @tagPattern = /(<(\/?))([^\s>]+)([\s>]|$)/
     @wordRegex = /[^>\r\n]*/
-    @tagSelector = SelectorCache.get('meta.tag | punctuation.definition.tag')
-    @commentSelector = SelectorCache.get('comment.*')
 
   patternForTagName: (tagName) ->
     tagName = _.escapeRegExp(tagName)
     new RegExp("(<#{tagName}([\\s>]|$))|(</#{tagName}>)", 'gi')
 
   isRangeCommented: (range) ->
-    scopes = @editor.scopeDescriptorForBufferPosition(range.start).getScopesArray()
-    @commentSelector.matches(scopes)
+    @scopesForPositionMatchRegex(range.start, COMMENT_SELECTOR_REGEX)
 
   isTagRange: (range) ->
-    scopes = @editor.scopeDescriptorForBufferPosition(range.start).getScopesArray()
-    @tagSelector.matches(scopes)
+    @scopesForPositionMatchRegex(range.start, TAG_SELECTOR_REGEX)
 
   isCursorOnTag: ->
-    @tagSelector.matches(@editor.getLastCursor().getScopeDescriptor().getScopesArray())
+    @scopesForPositionMatchRegex(@editor.getCursorBufferPosition(), TAG_SELECTOR_REGEX)
+
+  scopesForPositionMatchRegex: (position, regex) ->
```

Comment:
Yeah, I'm in favor of the change you've made here. When we have real trees we'll have a completely different code path for bracket matching and may want to just bake it into the editor.